src/coreclr/src/inc/corinfoinstructionset.h

```diff
@@ -0,0 +1,276 @@
+
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+// DO NOT EDIT THIS FILE! IT IS AUTOGENERATED
+// FROM /src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/InstructionSetDesc.txt
+// using /src/coreclr/src/tools/Common/JitInterface/ThunkGenerator/gen.bat
+
+#ifndef CORINFOINSTRUCTIONSET_H
+#define CORINFOINSTRUCTIONSET_H
+
+enum CORINFO_InstructionSet
+{
+    InstructionSet_ILLEGAL = 0,
+    InstructionSet_NONE = 63,
+#ifdef TARGET_ARM64
+    InstructionSet_ArmBase=1,
+    InstructionSet_ArmBase_Arm64=2,
+    InstructionSet_AdvSimd=3,
+    InstructionSet_AdvSimd_Arm64=4,
+    InstructionSet_Aes=5,
+    InstructionSet_Crc32=6,
+    InstructionSet_Crc32_Arm64=7,
+    InstructionSet_Sha1=8,
+    InstructionSet_Sha256=9,
+    InstructionSet_Atomics=10,
+    InstructionSet_Vector64=11,
+    InstructionSet_Vector128=12,
+#endif // TARGET_ARM64
+#ifdef TARGET_AMD64
+    InstructionSet_SSE=1,
+    InstructionSet_SSE2=2,
+    InstructionSet_SSE3=3,
+    InstructionSet_SSSE3=4,
+    InstructionSet_SSE41=5,
+    InstructionSet_SSE42=6,
+    InstructionSet_AVX=7,
+    InstructionSet_AVX2=8,
```

Comment:
Not needed at this point (still plenty of room in the bitmask), but for instruction sets that are implied by other instruction sets, a better strategy could be used here to pack these bits.  For instance, for x86 SIMD, a "SIMD level" thing could be used (e.g. `InstructionSet_SIMD_Level >= InstructionSet_SSE42` to test if SSE 4.2 is supported, with `InstructionSet_SIMD_Level` being in this case a 3-bit mask).

Reply:
Agreed, one thing to note is that the representation as a 64bit bitmask is an internal implementation detail, and we can change it fairly easily when we reach to having more than 62 instruction sets on a specific architecture. I believe that this scheme should last for at least 5 more years, which is long enough I don't care to plan further ahead.